usr/libexec/greenboot/greenboot

```diff
@@ -7,6 +7,53 @@ SCRIPTS_CHECK_PATHS=("/usr/lib/greenboot/check" "/etc/greenboot/check")
 SCRIPTS_GREEN_PATHS=("/usr/lib/greenboot/green.d" "/etc/greenboot/green.d")
 SCRIPTS_RED_PATHS=("/usr/lib/greenboot/red.d" "/etc/greenboot/red.d")
 
+source_configuration_file() {
+    GREENBOOT_CONFIGURATION_FILE=/etc/greenboot/greenboot.conf
+    if test -f "$GREENBOOT_CONFIGURATION_FILE"; then
+        source $GREENBOOT_CONFIGURATION_FILE
+    fi
+}
+
+source_configuration_file
+function exists_in_list {
+    VALUE=$1
+    echo "${DISABLED_HEALTHCHECKS,,}" | tr " " '\n' | grep -F -q -x "$VALUE"
+}
```

Comment:
```suggestion
function is_disabled {
    HEALTH_CHECK=$1    
    for DISABLED_HEALTH_CHECK in "${DISABLED_HEALTH_CHECKS[@]}"; do
        [ "${HEALTH_CHECK}" != "${DISABLED_HEALTH_CHECK}" ] || return true
    done
    return false     
}
```
You don't need to rely on external commands to iterate an array

Reply:
That's fair, I liked the `tr` and `grep` solution for its compactness. I agree that the looping solution you provided is easier to follow, though. (The `grep` solution also made me feel smart for all the fancy flags, but don't tell anyone that.)

Reply:
Rewriting this to be more like what you have above and I just want to make sure I'm understanding it. In what you have above `HEALTHCHECK` does not equal `DISABLED_HEALTHCHECK` returns true, which in the context of the function means "yes it is disabled"? Shouldn't this be reversed, as `HEALTHCHECK` not equaling `DISABLED_HEALTHCHECK` would mean that it isn't in the list, and therefore not disabled? I think I am just confusing myself now.

Reply:
The `||` operator in bash works this way: 
- It runs the command on the left side and if the returned code was `0` (the command finished successfully) then it skips the command on the right side and continues the execution (in this case it will continue with the next loop). 
- In case the return code is different than `0` (the command failed for whatever reason) then it runs the command on the right side.

So as long as `HEALTHCHECK` is different for the `DISABLED_HEALTHCHECK` the comparison (`[ "${HEALTHCHECK}" != "${DISABLED_HEALTHCHECK}" ]`) will be `true` so the command on the right side will be ignored and the for loop will keep running (we ignore the disabled health checks which are not equal to the healthcheck we are looking for).

On the other hand, if `HEALTHCHECK`  is equal to `DISABLED_HEALTHECK` (we found that this health check is disabled) then the comparison will be `false`, the command on the right side is executed and the `is_disabled` function will return `true`. 

In case the `for` loop ends, it means we didn't found the `HEALTCHECK` on the list (so it is not disabled) and the `is_disabled` function returns `false`.

Reply:
Thank you for the explanation!! I have never used that operator in bash before, I thought it was a shorthand if statement (which I guess it sort of is, just not how I was picturing it). I rewrote it to be a one line if statement because that made more sense to me, but if I get a chance I'll try to rework it with the `||`. Thanks again!

Reply:
Addressed in the most recent updates, I was having issues getting the `||` to work though so I stuck with the if statement. Leaving this open so I remember to keep finagling with the `||`